src/remote_cluster.rs

```diff
@@ -3,18 +3,19 @@ use crate::{
     DEFAULT_ANALYTICS_TIMEOUT, DEFAULT_DATA_TIMEOUT, DEFAULT_MANAGEMENT_TIMEOUT,
     DEFAULT_QUERY_TIMEOUT, DEFAULT_SEARCH_TIMEOUT, DEFAULT_TRANSACTION_TIMEOUT,
 };
+use serde_derive::{Deserialize, Serialize};
 use std::sync::{Arc, Mutex};
 use std::time::Duration;
 
-#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[derive(Debug, Deserialize, Serialize, Clone, Copy, Eq, PartialEq)]
 pub enum RemoteClusterType {
     Provisioned,
     Other,
 }
 
 impl From<Vec<String>> for RemoteClusterType {
     fn from(hostnames: Vec<String>) -> Self {
-        if hostnames.len() == 1 && hostnames[0].contains(&CAPELLA_SRV_SUFFIX.to_string()) {
+        if hostnames.len() == 1 && (hostnames[0].contains(&CAPELLA_SRV_SUFFIX.to_string())) {
```

Comment:
I'm assuming this code only gets run if a cluster_type isn't specified in the config?

Reply:
It always gets run but the result is ignored if the cluster_type has been specified